lib/kindle_highlights/highlight.rb

```diff
@@ -1,18 +1,22 @@
 module KindleHighlights
   class Highlight
-    attr_accessor :asin, :text, :location
+    attr_accessor :asin, :text, :note, :page, :location
 
     def self.from_html_elements(book:, html_elements:)
       new(
         asin: book.asin,
         text: html_elements.children.search("div.kp-notebook-highlight").first.text.squish,
+        note: html_elements.children.search("span#note").first.text,
+        page: html_elements.children.search("span#annotationHighlightHeader").first.text.partition('|').last.lstrip,
```

Comment:
Since this attribute is already named `page`, what do you think about just returning the page number as an integer? Instead of `"Page: 7"`, just `7`.

Reply:
We could do this - it's just a bit more complicated as we'd then need to read the full string and only return the page if the word is page... but happy to do so.  What I was looking for was a generalised location which preferentially gave me the page number if available, and fell back to the location if it wasn't.  This was also programmatically much easier!  However I can see it's a bit limiting.  I'll have a go at reworking it to just return the page number, and can rebuild the Page/Location from the two bits in the example.

Reply:
I've rewritten this now.